### PR TITLE
support Gitlab self hosted instance (#89)

### PR DESCRIPTION
**Pull request by xuchaoying**
**What**: allow user to input private_token to authenticate with the GitLab API

**Why**: self hosted Gitlab instance not working due to the api failing #89 

**How**: basically add a question to ask for user private toekn, and append the private token as another param to the gitlab apis.

**Checklist**:

- [x] Documentation
- [x] Tests
- [x] Ready to be merged 
- [x] Added myself to contributors table

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -158,6 +158,7 @@ These are the keys you can specify:
   --> `all-contributors-cli`. Mandatory.
 * `repoType`: Type of repository. Must be either `github` or `gitlab`. Default: `github`.
 * `repoHost`: Points to the repository hostname. Change it if you use a self hosted repository. Default: `https://github.com` if `repoType` is `github`, and `https://gitlab.com` if `repoType` is `gitlab`.
+* `private_token`: The personal access token to to authenticate with the GitLab API. Offer it if you use a self hosted repository. Default: `''`.
```
Comment (kentcdodds) [inline README.md]: Could this follow the existing camel casing convention? It should be `privateToken`.

**Comment by hipstersmoothie** [comment]
@xuchaoying I am going to be opening a PR to make this repo work with Github Enterprise. The private token code is needed! I will reference this PR in mine.

**Comment by chris-dura** [comment]
really looking forward to getting this merged in...

**Comment by kentcdodds** [comment]
Sorry, there are a bunch of merge conflicts here that need to be resolved :-/

**Comment by xuchaoying** [comment]
sorry for the belated commit. :( 

Private token needs to be set as an environment variable now. @kentcdodds

File: README.md [hunk 2]
```diff
@@ -137,6 +137,18 @@ Where `username` is the user's GitHub or Gitlab username, and `contribution` is
...
+Please note that if you are using a self-hosted gitlab instance, before add 
+contributor, you need to set an environment variable named `PRIVATE_TOKEN` first.
+
+> Private token is the personal access token to authenticate with the GitLab API.
+
+```console
+# set private token on linux
+export PRIVATE_TOKEN=your_privete_token
+# set private token on windows
+set PRIVATE_TOKEN=your_privete_token
```
Comment (chris-dura) [inline README.md]: `privete` -> `private`

File: package.json [hunk 3]
```diff
@@ -5,7 +5,9 @@
   "bin": {
     "all-contributors": "dist/cli.js"
   },
-  "files": ["dist"],
+  "files": [
+    "dist"
+  ],
```
[the remark this answers is not in the document]
Comment (chris-dura) [reply package.json]: I believe these changes are generated by `prettier` automatically on commit (?), and prettier hadn't been run in [awhile](https://github.com/jfmengels/all-contributors-cli/pull/95#pullrequestreview-125208168). So, these whitespace changes are probably the correct format.

[the remark this answers is not in the document]
Comment (xuchaoying) [reply package.json]: @chris-dura yes, maybe it's prettier. I didn't change the format at all.

**Pull request merged**